From f2ed466056b78748284aab2fe3a4e828c098f95d Mon Sep 17 00:00:00 2001
From: Joel Winarske <joel.winarsk@gmail.com>
Date: Tue, 14 Mar 2023 18:16:10 +0000
Subject: [PATCH] Do not configure psimd

Signed-off-by: Joel Winarske <joel.winarsk@gmail.com>
---
 CMakeLists.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33edb53..b74c346 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,11 +118,11 @@ INSTALL(TARGETS fp16
   PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 # ---[ Configure psimd
-IF(NOT TARGET psimd)
-  ADD_SUBDIRECTORY(
-    "${PSIMD_SOURCE_DIR}"
-    "${CMAKE_BINARY_DIR}/psimd")
-ENDIF()
+# IF(NOT TARGET psimd)
+#  ADD_SUBDIRECTORY(
+#    "${PSIMD_SOURCE_DIR}"
+#    "${CMAKE_BINARY_DIR}/psimd")
+# ENDIF()
 
 IF(FP16_BUILD_TESTS)
   # ---[ Build google test
-- 
2.39.2

